app/models/course/assessment.rb

```diff
@@ -13,6 +13,9 @@ class Course::Assessment < ActiveRecord::Base
   after_initialize :set_defaults, if: :new_record?
   before_validation :propagate_course
   before_validation :assign_folder_attributes
+  before_validation :prevent_reminder_duplication
+  before_save :setup_opening_reminders, if: :start_at_changed?, unless: :draft?
```

Comment:
The start_at_changed cannot be detected in after_save, due to acts_as. Before_save is used in this case to register the changes, and job will only be queue after_commit.

There's one more layer of check in the job itself, which ensure that the input start_at date is consistent with the model start_at date.